include/bl_storage.h

```diff
@@ -50,6 +53,7 @@ u32_t num_public_keys_read(void);
  * @retval -EINVAL  Key has been invalidated.
  * @retval -ENOMEM  The provided buffer is too small.
  * @retval -EFAULT  key_idx is too large. There is no key with that index.
+ * @retval -EHASHFF If any key data contains 0xFFFF, which is unsupported.
```

Comment:
Should we document whether or not returning `0` is an error? The return value for this  function is checked with `read_retval < 0` to indicate failure, but should we treat 0 as an error as well?

Reply:
Changed bl_validation to check for the correct length directly. I couldn't think of a good way to document 0, so I left it as is.